ksmt-symfpu/src/main/kotlin/org/ksmt/symfpu/solver/SymfpuSolver.kt

```diff
@@ -0,0 +1,59 @@
+package org.ksmt.symfpu.solver
+
+import org.ksmt.KContext
+import org.ksmt.decl.KConstDecl
+import org.ksmt.expr.KExpr
+import org.ksmt.solver.KModel
+import org.ksmt.solver.KSolver
+import org.ksmt.solver.KSolverConfiguration
+import org.ksmt.solver.KSolverStatus
+import org.ksmt.sort.KBoolSort
+import kotlin.time.Duration
+
+open class SymfpuSolver<Config : KSolverConfiguration>(
+    val solver: KSolver<Config>,
+    val ctx: KContext,
+) : KSolver<Config> {
+
+    private val transformer = FpToBvTransformer(ctx)
+    private val mapTransformedToOriginalAssertions =
+        mutableMapOf<KExpr<KBoolSort>, KExpr<KBoolSort>>()
+
+    override fun configure(configurator: Config.() -> Unit) {
+        solver.configure(configurator)
+    }
+
+    override fun assert(expr: KExpr<KBoolSort>) = solver.assert(transformer.applyAndGetExpr(expr)) // AndGetExpr
+
+    override fun assertAndTrack(expr: KExpr<KBoolSort>, trackVar: KConstDecl<KBoolSort>) =
+        solver.assertAndTrack(transformer.applyAndGetExpr(expr).also { mapTransformedToOriginalAssertions[it] = expr },
```

Comment:
It is better to avoid complex expressions in call arguments. To make this code more clear we can use named arguments and put args on separate lines:
```kotlin
solver.assertAndTrack(
    expr = transformer.applyAndGetExpr(expr).also { mapTransformedToOriginalAssertions[it] = expr },
    trackVar = trackVar
)
```